src/state/service.rs

```diff
@@ -94,18 +101,18 @@ pub struct ServiceStore {
     /// Allows for lookup of services by network address, the service's xds secondary key.
     /// this map stores the same services as `services_by_hostname` so we can mutate the services in both maps
     /// at once when a service is updated or workload updates affect a service's endpoints.
-    by_vip: HashMap<NetworkAddress, Arc<RwLock<Service>>>,
+    by_vip: HashMap<NetworkAddress, Arc<Service>>,
 
     /// Allows for lookup of services by namespaced and hostname, the service's xds primary key.
     /// this map stores the same services as `services_by_ip` so we can mutate the services in both maps
     /// at once when a service is updated or workload updates affect a service's endpoints.
-    by_host: HashMap<String, Vec<Arc<RwLock<Service>>>>,
+    by_host: HashMap<String, Vec<Arc<Service>>>,
```

Comment:
can we make `by_host` take a `NamespacedHostname` rather than just a string? I'd prefer compilation errors to confusing runtime ones if we try to send a malformed key

Reply:
This is a change required by #536. The DNS proxy only has the requested hostname.

Reply:
can we update the comment to correctly reflect that this is no longer keyed by namespaced hostname then?